app/code/core/Mage/Adminhtml/Block/Widget/Form/Element/Dependence.php

```diff
@@ -47,35 +53,193 @@ class Mage_Adminhtml_Block_Widget_Form_Element_Dependence extends Mage_Adminhtml
+     *     'source_1' => 'foo',
+     *     'source_2' => 'bar',
+     * ]);
+     *
+     * Example 4: `result` field will be shown if `source_1 == 'foo' XOR source_2 == 'bar'`
+     * If more than two conditions are provided, returns true if exactly one condition is true
+     *
+     * $block->addComplexFieldDependence('result', $block::MODE_XOR, [
+     *     'source_1' => 'foo',
+     *     'source_2' => 'bar',
+     * ]);
+     *
+     * Example 5: `result` field will be shown if `(source_1 == 'foo') OR (source_2 == 'bar' AND source_3 == 'baz')`
+     *
+     * $block->addComplexFieldDependence('result', $block::MODE_OR, [
+     *     'source_1' => 'foo',
+     *     $block->createCondition($block::MODE_AND, [
+     *         'source_2' => 'bar',
+     *         'source_3' => 'baz',
+     *     ]),
+     * ]);
+     *
+     * @param string $targetField field to be toggled
+     * @param self::MODE_* $operator one of MODE_NOT, MODE_AND, MODE_OR, MODE_XOR
+     * @param array $condition
      * @return $this
      */
-    public function addFieldDependence($fieldName, $fieldNameFrom, $refValues)
+    public function addComplexFieldDependence(string $targetField, string $operator, array $condition): self
     {
-        $this->_depends[$fieldName][$fieldNameFrom] = $refValues;
+        $this->_depends[$targetField][] = $this->createCondition($operator, $condition);
         return $this;
     }
 
     /**
-     * Add misc configuration options to the javascript dependencies controller
+     * Return a field's full simple or complex field dependence condition
+     */
+    public function getRawFieldDependence(string $targetField): ?array
+    {
+        return $this->_depends[$targetField] ?? null;
+    }
+
+    /**
+     * Set a field's full simple or complex field dependence condition
+     */
+    public function setRawFieldDependence(string $targetField, array $condition): self
+    {
+        // Complex conditions must be wrapped in an array
+        if ($this->isLogicalOperator($condition['operator'] ?? null) && is_array($condition['condition'] ?? null)) {
+            $condition = [$condition];
+        }
+        $this->_depends[$targetField] = $condition;
+        return $this;
+    }
+
+    /**
+     * Clear a field's full simple or complex field dependence condition
+     */
+    public function clearFieldDependence(string $targetField): self
+    {
+        unset($this->_depends[$targetField]);
+        return $this;
+    }
+
+    /**
+     * Add field alias to id mapping
+     *
+     * @param string $fieldId element ID in DOM
+     * @param string $fieldAlias alias to refer to this field by
+     * @return $this
+     */
+    public function addFieldMap($fieldId, $fieldAlias)
+    {
+        $this->_fields[$fieldAlias] = $fieldId;
+        return $this;
+    }
+
+    /**
+     * Add configuration option to the javascript dependencies controller
+     *
+     * @see self::addConfigOptions()
```

Comment:
```suggestion
```

Reply:
I only put that there because `addConfigOptions` is where the config options are documented, but I don't know how it shows up in PHPStorm, so feel free to commit the suggestion.